src/battle_main.c

```diff
@@ -299,27 +299,220 @@ const struct OamData gOamData_BattleSpritePlayerSide =
 
 static const s8 sCenterToCornerVecXs[8] ={-32, -16, -16, -32, -32};
 
-const u8 gTypeNames[NUMBER_OF_MON_TYPES][TYPE_NAME_LENGTH + 1] =
+#define GENERIC(type) .generic = _(#type" move")
+#define MOVES(z, max)               \
+        .zMove = MOVE_##z,          \
+        .maxMove = MOVE_MAX_##max,
+
+#define ITEMS(type, item, berryName, crystal) \
+        .enhanceItem = ITEM_##item,           \
+        .berry = ITEM_##berryName##_BERRY,    \
+        .gem = ITEM_##type##_GEM,             \
+        .zCrystal = ITEM_##crystal##IUM_Z,
+
+//Not in ITEMS since Normal doesn't have these two items (Blank Plate is not yet implemented)
+#define LEGENDARY_ITEMS(plateName, type)   \
+        .plate = ITEM_##plateName##_PLATE, \
+        .memory = ITEM_##type##_MEMORY,    \
+
+//Not in ITEMS since Stellar has Tera Shards but no other item
+#define TERA_SHARD(type) .teraShard = ITEM_##type##_TERA_SHARD,
+
+// .generic is large enough that the text for TYPE_ELECTRIC will exceed TEXT_BUFF_ARRAY_COUNT.
+const struct Type gTypes[NUMBER_OF_MON_TYPES] =
 {
-    [TYPE_NORMAL] = _("Normal"),
-    [TYPE_FIGHTING] = _("Fight"),
-    [TYPE_FLYING] = _("Flying"),
-    [TYPE_POISON] = _("Poison"),
-    [TYPE_GROUND] = _("Ground"),
-    [TYPE_ROCK] = _("Rock"),
-    [TYPE_BUG] = _("Bug"),
-    [TYPE_GHOST] = _("Ghost"),
-    [TYPE_STEEL] = _("Steel"),
-    [TYPE_MYSTERY] = _("???"),
-    [TYPE_FIRE] = _("Fire"),
-    [TYPE_WATER] = _("Water"),
-    [TYPE_GRASS] = _("Grass"),
-    [TYPE_ELECTRIC] = _("Electr"),
-    [TYPE_PSYCHIC] = _("Psychc"),
-    [TYPE_ICE] = _("Ice"),
-    [TYPE_DRAGON] = _("Dragon"),
-    [TYPE_DARK] = _("Dark"),
-    [TYPE_FAIRY] = _("Fairy"),
+    [TYPE_NORMAL] =
+    {
+        .name = _("Normal"),
+        GENERIC(a NORMAL),
```

Comment:
I may be alone here, but I find it very weird that this string isn't in clear quotation marks. I know the macro deals with it, but it just feels wrong to me